src/_pytest/python.py

```diff
@@ -594,6 +580,15 @@ def isinitpath(self, path):
         return path in self.session._initialpaths
 
     def collect(self):
+        # XXX: HACK!
```

Comment:
Strange, I expect `main.pytest_ignore_collect` to handle this properly already:

https://github.com/pytest-dev/pytest/blob/4d8903fd0b89369e0e44f084e93c43d2ce26d736/src/_pytest/main.py#L261-L284

If `pytest_ignore_collect` sees the same path twice, it should ignore the collection when see the path the second time, which would not trigger a new `Package` item to be collected.

Reply:
The problem is it sees that some modules were already "collected" (at Session level) via the recurse() method. Then when the package goes and tries to collect its children and turn them into nodes it can't because they look like duplicates. Wrong! My first reaction was to try to stop marking them as duplicates at Session level, but everything is so tangled up that I can't find a cleaner way.

Reply:
JFI: I've moved it in https://github.com/pytest-dev/pytest/pull/4241.